app/models/storage_service.rb

```diff
@@ -0,0 +1,21 @@
+class StorageService < ApplicationRecord
+  include NewWithTypeStiMixin
+  include ProviderObjectMixin
+  include AsyncDeleteMixin
+  include AvailabilityMixin
+  include SupportsFeatureMixin
+  include CustomActionsMixin
+
+  belongs_to :ext_management_system, :foreign_key => :ems_id,
+                                     :class_name  => "ExtManagementSystem"
+  has_many :storage_service_resource_attachments, :inverse_of => :storage_service, :dependent => :destroy
+  has_many :storage_resources, :through => :storage_service_resource_attachments
+
```

Comment:
You need `has_many :cloud_volumes` in this model

```
>> cv.storage_service = cinder.storage_services.first
ActiveRecord::InverseOfAssociationNotFoundError (Could not find the inverse association for storage_service (:cloud_volumes in StorageService))
```

Reply:
I wonder how we have missed this.

Reply:
No worries it happens :smile: that is why we review